File: Codinsa2015.Client/Codinsa2015.Client/Views/Clank/ClientCpp/src/StateAlterationView.cpp
#include "../inc/StateAlterationView.h"
void StateAlterationView::serialize(std::ostream& output) {
	// Source
	output << ((int)this->Source) << '\n';
	// SourceType
	output << ((int)this->SourceType) << '\n';
	// Model
	this->Model.serialize(output);
	// Parameters
	this->Parameters.serialize(output);
	// RemainingTime
	output << ((float)this->RemainingTime) << '\n';
}

StateAlterationView StateAlterationView::deserialize(std::istream& input) {
	StateAlterationView _obj = StateAlterationView();
	// Source
	int _obj_Source; input >> _obj_Source; input.ignore(1000, '\n');
	_obj.Source = (int)_obj_Source;
	// SourceType
	int _obj_SourceType; input >> _obj_SourceType; input.ignore(1000, '\n');
	_obj.SourceType = (::StateAlterationSource)_obj_SourceType;
	// Model
	StateAlterationModelView _obj_Model = StateAlterationModelView::deserialize(input);
	_obj.Model = (::StateAlterationModelView)_obj_Model;
	// Parameters
	StateAlterationParametersView _obj_Parameters = StateAlterationParametersView::deserialize(input);
	_obj.Parameters = (::StateAlterationParametersView)_obj_Parameters;
	// RemainingTime
	float _obj_RemainingTime; input >> _obj_RemainingTime; input.ignore(1000, '\n');
	_obj.RemainingTime = (float)_obj_RemainingTime;
	return _obj;
}

File: Codinsa2015.Client/Codinsa2015.Client/Views/Clank/ClientJava/src/StateAlterationView.java
import java.lang.*;

	public class StateAlterationView
	{

	
		public int Source;	
		public StateAlterationSource SourceType;	
		public StateAlterationModelView Model;	
		public StateAlterationParametersView Parameters;	
		public float RemainingTime;	
		public static StateAlterationView Deserialize(BufferedReader input) {
		try {
			StateAlterationView _obj =  new StateAlterationView();
			// Source
			int _obj_Source = Integer.Parse(input.readLine());
			_obj.Source = _obj_Source;
			// SourceType
			int _obj_SourceType = Integer.Parse(input.readLine());
			_obj.SourceType = StateAlterationSource.fromValue(_obj_SourceType);
			// Model
			StateAlterationModelView _obj_Model = StateAlterationModelView.deserialize(input);
			_obj.Model = _obj_Model;
			// Parameters
			StateAlterationParametersView _obj_Parameters = StateAlterationParametersView.deserialize(input);
			_obj.Parameters = _obj_Parameters;
			// RemainingTime
			float _obj_RemainingTime = Float.Parse(input.readLine());
			_obj.RemainingTime = _obj_RemainingTime;
			} catch (UnsupportedEncodingExceptio e) { 
			} catch (IOException e) { }
			return _obj;
		}

		public void serialize(OutputStreamWriter output) {
			try {
			// Source
			output.append(((Integer)this.Source).toString() + "\n");
			// SourceType
			output.append(((Integer)(this.SourceType.getValue())).toString() + "\n");
			// Model
			this.Model.serialize(output);
			// Parameters
			this.Parameters.serialize(output);
			// RemainingTime
			output.WriteLine(((Float)this.RemainingTime).toString() + "\n");
			} catch (UnsupportedEncodingExceptio e) { 
			} catch (IOException e) { }
		}

	}
}
